redisson/src/main/java/org/redisson/command/RedisQueuedBatchExecutor.java

```diff
@@ -100,7 +100,10 @@ protected void releaseConnection(CompletableFuture<R> attemptPromise, Completabl
                 || RedisCommands.DISCARD.getName().equals(command.getName())) {
             if (attempt < attempts
                     && attemptPromise.isCompletedExceptionally()) {
-                return;
+                Throwable cause = cause(attemptPromise);
```

Comment:
Can you use just replace `attemptPromise.isCompletedExceptionally()` with `attemptPromise.isCancelled()` ?

Reply:
my bad. resubmitted.